### DISCUSSION TITLE
Berry Script: How to list active timers or active crons?

### DISCUSSION THREAD

**Discussion by ozett** [arendst/Tasmota#17913] [Q&A]
Hi,
hope discission/Questions is the right place for this, it is not an issue.

But as i am experimenting with berry, i was wondering
a)
how to list active timers in berry (to remove them? one or all)?
b)
also the same with cron entries. How to list - to remove them, one or all?

thanks for help,

**Comment by sfromis** [comment]
There is no feature for querying active timers/crons. For debugging purposes, you could check the internal lists `tasmota._timers` or `tasmota._crons`, but this is absolutely not meant as some sort of API, and data may change without this counting as a breaking change.

What I generally do with timers/crons is to include an id, a safe strategy for cleanup is then to remove all the id's the particular application knows about, as it should definitely not try stunts like removing timers/crons not created by itself.
